addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/pscan/WebSocketPassiveScanThread.java

```diff
@@ -0,0 +1,180 @@
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.websocket.pscan;
+
+import java.util.Iterator;
+import java.util.concurrent.LinkedBlockingQueue;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.db.DatabaseException;
+import org.zaproxy.zap.extension.websocket.WebSocketMessage;
+import org.zaproxy.zap.extension.websocket.WebSocketMessageDTO;
+import org.zaproxy.zap.extension.websocket.WebSocketObserver;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy;
+import org.zaproxy.zap.extension.websocket.alerts.WebSocketAlertThread;
+import org.zaproxy.zap.extension.websocket.alerts.WebSocketAlertWrapper;
+import org.zaproxy.zap.extension.websocket.db.TableWebSocket;
+import org.zaproxy.zap.extension.websocket.db.WebSocketStorage;
+
+/** Implements a background thread for passive scanning */
+public class WebSocketPassiveScanThread extends Thread
+        implements WebSocketObserver, WebSocketAlertThread {
+
+    private static final Logger LOGGER = Logger.getLogger(WebSocketPassiveScanThread.class);
+
+    private static final int WEBSOCKET_OBSERVING_ORDER =
+            WebSocketStorage.WEBSOCKET_OBSERVING_ORDER + 10;
+
+    /** Interval for message reading */
+    public static final int SLEEP_TIME = 5000;
+
+    /** Thread safe Linked Queue */
+    private LinkedBlockingQueue<MessageWrapper> messagesBuffer;
+
+    /** {@code True} to enable the passive scan Thread */
+    private volatile boolean isActive;
+
+    /** Reference to Database. Used in order to pick messages for scanning */
+    private TableWebSocket tableWebSocket;
+
+    /**
+     * Manager used to updating the messages table. In addition, used by passive scan thread so as
+     * to get access at plugins
+     */
+    private WebSocketPassiveScannerManager passiveScannerManager;
+
+    /**
+     * Initialize the passive scan in background thread. By default thread is inactive and not
+     * alive. In order to activate thread set {@link WebSocketPassiveScanThread#setActive(boolean)}
+     * true and start thread with {@link WebSocketPassiveScanThread#start()}.
+     *
+     * @param passiveScannerManager the manager
+     */
+    public WebSocketPassiveScanThread(WebSocketPassiveScannerManager passiveScannerManager) {
```

Comment:
We should call`super("ZAP-WS-PassiveScanner");`(?) to have a descriptive name.

Reply:
Fixed in 734e0d9